version/distro/distro.go

```diff
@@ -93,8 +109,32 @@ func linuxDistro() Distro {
 		return WDMyCloud
 	case have("/etc/unraid-version"):
 		return Unraid
+	default:
+		if d, err := parseLinuxOSRelease("/etc/os-release"); err == nil {
+			return d
+		}
+		return Unknown
 	}
-	return ""
+}
+
+func parseLinuxOSRelease(path string) (Distro, error) {
+	// See https://www.freedesktop.org/software/systemd/man/os-release.html
+	f, err := os.Open(path)
+	if err != nil {
+		return Unknown, err
+	}
+	defer f.Close()
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if !strings.HasPrefix(s.Text(), "ID=") {
+			continue
+		}
+		d := strings.TrimPrefix(s.Text(), "ID=")
```

Comment:
this looks like two allocs on match

it could be one with CutPrefix.

but then it'd still allocate on each line that didn't match. but you could use `!mem.HasPrefix(s.Bytes(), mem.S("ID="))` and not allocate on the non-matching lines

Reply:
Done.
I've never used that mem package, please correct if my usage is wrong